adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Helpers/EncodingHelper.cs

```diff
@@ -53,8 +55,31 @@ public static string UrlEncode(string message)
                 return message;
             }
 
-            message = Uri.EscapeDataString(message);
-            message = message.Replace("%20", "+");
+            if (message.Length < MaxUrlEncodingSize)
+            {
+                message = Uri.EscapeDataString(message);
+                message = message.Replace("%20", "+");
+            }
+            else
+            {
+                StringBuilder sb = new StringBuilder();
```

Comment:
Why is this optimization needed? Perhaps add a comment. Also - for simplicity, why not get rid of the simple way ? (lines 60-61)

Reply:
I was considering the same... but then if we call this method a lot of times especially on service scenarios, it might make sense... but absolutely a good idea to have that comment ensuring others will not optimize it later.

Reply:
The original way would be slightly more performant and does not change for existing scenarios. I will add a comment on this. Thanks!